meerk40t/gui/scene/scenewidgets.py

```diff
@@ -583,12 +583,38 @@ class RectSelectWidget(Widget):
     Rectangle Selection Widget, draws the selection rectangle if left-clicked and dragged
     """
 
+    # selection_method = 1 = hit, 2 = cross, 3 = enclose
+    # Color for selection rectangle (hit, cross, enclose)
+    selection_colors = [
+        wx.RED,
+        wx.GREEN,
+        wx.BLUE,
+    ]
+    # Visual help for selection rectangle (to help those with visual challenges i.e colorblindness)
+    selection_line = [wx.PENSTYLE_DOT_DASH, wx.PENSTYLE_DOT, wx.PENSTYLE_SHORT_DASH]
+    selection_text = [
+        "Select all elements that the selection rectangle touches",
+        "Select all elements that the selection rectangle crosses in at least one dimension",
+        "Select all elements that the selection rectangle fully encloses",
+    ]
```

Comment:
1. Please translate these strings i.e. `_("text")` rather than `"text"` .
2. I like that these strings exist. Check the wxmain.py file as there is now a method by which you can send this to the status bar when the selection box is started and you can determine the type of selection based on the direction of movement (though I guess this can change if you start going NW and then go East and turn it into a NE drag. 
3. Can we please have versions for Control and Shift (which need to react if these are pressed or released whilst left-down?
4. There are three strings here, but I can see only two modes (RTL and LTR) when I test.

Reply:
> There are three strings here, but I can see only two modes (RTL and LTR) when I test.
The current test does no longer use cross-selecting, as the feedback so far indicates that touch is more intuitive.

Reply:
> The current test does no longer use cross-selecting, as the feedback so far indicates that touch is more intuitive.

That's fine - we either need to leave the code to do the cross-select in the code base but commented out (if we think we might ever want to use it) or remove all traces including the string.

Reply:
>In addition to the above comments, there is one major functional issue: in the enclosed test file, try selecting nothing >and then a red RTL selection of the middle emblem. Instead of it selecting every element that the red-selection box >touches in the emblem, it selects everything and this doesn't feel right to me.
Actually it does what it is supposed to do ;-) It selects all these elements but as well as it overlaps with the big box outside that one as well, so the biggest surrounding box is used to highlight the selection.
On second thought i think it would be better if we ignore selections fully inside a bigger object - have changed the code accordingly
I beleive though it might be more log

Reply:
I'll wait to see the code to see if it is a good algorthm.